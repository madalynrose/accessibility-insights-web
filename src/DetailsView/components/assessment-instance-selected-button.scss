// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';
@import '../../common/styles/common.scss';

.instance-visibility-button,
.instance-visibility-button:active,
.instance-visibility-button:focus {
    color: $primary-text;
    padding-bottom: 6px;
    background-color: inherit;
    @include ms-high-contrast-override {
        color: inherit;
        outline: 1px inherit solid;
    }
}

.test-instance-selected {
    padding: 2px;
    color: $neutral-0;
}

.test-instance-selected-hidden {
    background-color: $neutral-60;
    cursor: default;
}

.test-instance-selected-visible {
    background-color: $communication-primary;
}

.test-instance-selected-hidden-button {
    background-color: transparent;
}
