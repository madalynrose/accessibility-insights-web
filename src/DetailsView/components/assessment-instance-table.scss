// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

.assessment-instance-table {
    width: 100%;
}
