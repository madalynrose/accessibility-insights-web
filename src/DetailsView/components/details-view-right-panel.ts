// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentsFeatureFlagFilter } from 'assessments/assessments-feature-flag-filter';
import { AssessmentsRequirementsFilter } from 'assessments/assessments-requirements-filter';
import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { DetailsViewRightContentPanelType } from 'common/types/store-data/details-view-right-content-panel-type';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { OverviewSummaryReportModel } from 'reports/assessment-report-model';
import { GetAssessmentSummaryModelFromProviderAndStoreData } from 'reports/get-assessment-summary-model';
import { GetQuickAssessSummaryModelFromProviderAndStoreData } from 'reports/get-quick-assess-summary-model';
import { ReactFCWithDisplayName } from '../../common/react/named-fc';
import { DetailsViewPivotType } from '../../common/types/store-data/details-view-pivot-type';
import {
    getOverviewTitle,
    getTestViewTitle,
    GetTestViewTitleProps,
} from '../handlers/get-document-title';
import {
    GetLeftNavSelectedKeyProps,
    getOverviewKey,
    getTestViewKey,
} from './left-nav/get-left-nav-selected-key';
import {
    OverviewContainer,
    OverviewContainerDeps,
    OverviewContainerProps,
} from './overview-content/overview-content-container';
import { TargetChangeDialogDeps } from './target-change-dialog';
import {
    TestViewContainer,
    TestViewContainerDeps,
    TestViewContainerProps,
} from './test-view-container';

export type DetailsViewContentDeps = OverviewContainerDeps &
    TestViewContainerDeps &
    TargetChangeDialogDeps;

export type RightPanelProps = Omit<TestViewContainerProps, 'deps'> &
    Omit<OverviewContainerProps, 'deps'> & {
        deps: OverviewContainerDeps | TestViewContainerDeps;
    };

export type GetFilteredProviderDeps = {
    assessmentsProvider: AssessmentsProvider;
    assessmentsProviderWithFeaturesEnabled: AssessmentsFeatureFlagFilter;
    assessmentsProviderForRequirements: AssessmentsRequirementsFilter;
    quickAssessRequirementKeys: string[];
};

export type GetFilteredProviderProps = {
    deps: GetFilteredProviderDeps;
    featureFlagStoreData: FeatureFlagStoreData;
};

export type GetSummaryModelFromStoreDataDeps = {
    assessmentsProvider: AssessmentsProvider;
    getAssessmentSummaryModelFromProviderAndStoreData: GetAssessmentSummaryModelFromProviderAndStoreData;
    getQuickAssessSummaryModelFromProviderAndStoreData: GetQuickAssessSummaryModelFromProviderAndStoreData;
    quickAssessRequirementKeys: string[];
};

export type GetSummaryModelFromStoreDataProps = {
    deps: GetSummaryModelFromStoreDataDeps;
    assessmentStoreData: AssessmentStoreData;
};

export type DetailsRightPanelConfiguration = Readonly<{
    RightPanel: ReactFCWithDisplayName<RightPanelProps>;
    RightPanelProps?: {
        getFilteredProvider?: (props: GetFilteredProviderProps) => AssessmentsProvider;
        getSummaryModelFromProviderAndStoreData?: (
            props: GetSummaryModelFromStoreDataProps,
        ) => OverviewSummaryReportModel;
    };
    GetTitle: (props: GetTestViewTitleProps) => string;
    GetLeftNavSelectedKey: (props: GetLeftNavSelectedKeyProps) => string;
    GetStartOverContextualMenuItemKeys: () => string[];
}>;

export type GetDetailsRightPanelConfiguration = (
    props: GetDetailsRightPanelConfigurationProps,
) => DetailsRightPanelConfiguration;
export type GetDetailsRightPanelConfigurationProps = {
    selectedDetailsViewPivot: DetailsViewPivotType;
    detailsViewRightContentPanel: DetailsViewRightContentPanelType;
};

const TestViewPanel: DetailsRightPanelConfiguration = {
    RightPanel: TestViewContainer,
    GetTitle: getTestViewTitle,
    GetLeftNavSelectedKey: getTestViewKey,
    GetStartOverContextualMenuItemKeys: () => ['assessment', 'test'],
};

const detailsViewOverviewConfiguration: {
    [key in DetailsViewPivotType]: DetailsRightPanelConfiguration;
} = {
    [DetailsViewPivotType.assessment]: {
        RightPanel: OverviewContainer,
        RightPanelProps: {
            getFilteredProvider: (props: GetFilteredProviderProps) =>
                props.deps.assessmentsProviderWithFeaturesEnabled(
                    props.deps.assessmentsProvider,
                    props.featureFlagStoreData,
                ),
            getSummaryModelFromProviderAndStoreData: (props: GetSummaryModelFromStoreDataProps) =>
                props.deps.getAssessmentSummaryModelFromProviderAndStoreData(
                    props.deps.assessmentsProvider,
                    props.assessmentStoreData,
                ),
        },
        GetTitle: getOverviewTitle,
        GetLeftNavSelectedKey: getOverviewKey,
        GetStartOverContextualMenuItemKeys: () => ['assessment'],
    },
    [DetailsViewPivotType.mediumPass]: {
        RightPanel: OverviewContainer,
        RightPanelProps: {
            getFilteredProvider: (props: GetFilteredProviderProps) =>
                props.deps.assessmentsProviderForRequirements(
                    props.deps.assessmentsProviderWithFeaturesEnabled(
                        props.deps.assessmentsProvider,
                        props.featureFlagStoreData,
                    ),
                    props.deps.quickAssessRequirementKeys,
                ),
            getSummaryModelFromProviderAndStoreData: (props: GetSummaryModelFromStoreDataProps) =>
                props.deps.getQuickAssessSummaryModelFromProviderAndStoreData(
                    props.deps.assessmentsProvider,
                    props.assessmentStoreData,
                    props.deps.quickAssessRequirementKeys,
                ),
        },
        GetTitle: getOverviewTitle,
        GetLeftNavSelectedKey: getOverviewKey,
        GetStartOverContextualMenuItemKeys: () => ['assessment'],
    },
    [DetailsViewPivotType.fastPass]: TestViewPanel,
};

export const GetDetailsRightPanelConfiguration: GetDetailsRightPanelConfiguration = (
    props: GetDetailsRightPanelConfigurationProps,
) => {
    if (props.detailsViewRightContentPanel === 'TestView') {
        return TestViewPanel;
    }
    return detailsViewOverviewConfiguration[props.selectedDetailsViewPivot];
};
