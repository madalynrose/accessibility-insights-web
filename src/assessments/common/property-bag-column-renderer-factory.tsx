// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { propertyBagColumnRenderer } from './property-bag-column-renderer';
import * as React from 'react';

export class PropertyBagColumnRendererFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: InstanceTableRow<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}

export class PropertyBagColumnRendererWithComputationFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: InstanceTableRow<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRendererWithComputation(item, configs);
        };
    }
}

export function propertyBagColumnRendererWithComputation<TPropertyBag extends ColumnValueBag>(
    item: InstanceTableRow<TPropertyBag>,
    configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
): JSX.Element {
    const mapper = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        if (item.instance.propertyBag == null) {
            return null;
        }
        if (config.compute == null) {
            return null;
        }
        if (
            config.neededPropertyBagValues.some(
                key => item.instance.propertyBag.hasOwnProperty(key) === false,
            ) &&
            config.defaultValue == null
        ) {
            return null;
        }
        return render(config, index);
    };

    const render = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        console.log(item.instance);
        const ValueComponent = config.compute(item.instance.propertyBag);
        console.log('get by calling compute', ValueComponent);
        return (
            <div key={`property-${index}`} className="property-bag-div">
                <span className="display-name">{`${config.displayName}: `}</span>
                <ValueComponent />
            </div>
        );
    };

    return <div className="property-bag-container">{configs.map(mapper)}</div>;
}
