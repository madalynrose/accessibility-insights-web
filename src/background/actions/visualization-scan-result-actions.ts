// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Action } from 'common/flux/action';
import { ScanCompletedPayload } from '../../injected/analyzers/analyzer';
import {
    AddTabbedElementPayload,
    AddTabStopInstancePayload,
    RemoveTabStopInstancePayload,
    UpdateTabStopInstancePayload,
    UpdateTabStopRequirementStatusPayload,
} from './action-payloads';

export class VisualizationScanResultActions {
    public readonly scanCompleted = new Action<ScanCompletedPayload<any>>();
    public readonly getCurrentState = new Action();
    public readonly disableIssues = new Action();
    public readonly addTabbedElement = new Action<AddTabbedElementPayload>();
    public readonly disableTabStop = new Action();
    public readonly updateTabStopsRequirementStatus = new Action<
        UpdateTabStopRequirementStatusPayload
    >();
    public readonly addTabStopInstance = new Action<AddTabStopInstancePayload>();
    public readonly updateTabStopInstance = new Action<UpdateTabStopInstancePayload>();
    public readonly removeTabStopInstance = new Action<RemoveTabStopInstancePayload>();
}
