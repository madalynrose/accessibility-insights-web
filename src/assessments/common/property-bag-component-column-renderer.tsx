// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { NamedFC } from 'common/react/named-fc';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { isEmpty } from 'lodash';
import * as React from 'react';

import { DictionaryStringTo } from 'types/common-types';
import { Term } from 'assessments/markup';
import { PropertyBagColumnRenderer } from 'assessments/common/property-bag-column-renderer';

export class PropertyBagComponentColumnRenderer<
    TPropertyBag extends ColumnValueBag,
> extends PropertyBagColumnRenderer<TPropertyBag> {
    private render = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        value: any,
        index: number,
    ) => {
        return (
            <div key={`property-${index}`} className="property-bag-div">
                <span className="display-name">{`${config.displayName}: `}</span>
                {this.renderValue(config, value)}
            </div>
        );
    };

    private mapper = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        if (this.item.instance.propertyBag == null) {
            return null;
        }
        const value = this.item.instance.propertyBag[config.propertyName];
        if (value == null && config.defaultValue == null) {
            return null;
        }

        return this.render(config, value, index);
    };

    protected renderValue = (config: PropertyBagColumnRendererConfig<TPropertyBag>, value: any) => {
        if (config.expand) {
            return this.renderProperties(config, value);
        }
        return <React.Fragment>{() => value}</React.Fragment>;
        // const valueToRender = value || config.defaultValue;
        // if (typeof valueToRender === 'string') {
        //     return <React.Fragment>{valueToRender}</React.Fragment>;
        // }

        // const ValueComponent = valueToRender;
        // console.log(ValueComponent, <ValueComponent />);
        // return (
        //     <React.Fragment>
        //         Hello, this is a <Emphasis>test</Emphasis>
        //     </React.Fragment>
        // );
    };

    private renderProperties = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        propertyMap: DictionaryStringTo<string>,
    ) => {
        if (isEmpty(propertyMap)) {
            return <React.Fragment>{config.defaultValue}</React.Fragment>;
        }

        return Object.keys(propertyMap).map(key => {
            return (
                <div key={key} className="expanded-property-div">
                    {this.renderInnerKeyValue(key, propertyMap[key])}
                </div>
            );
        });
    };

    private renderInnerKeyValue = (key: string, value: string) => {
        if (value === null) {
            return <span className="display-name">{`${key}`}</span>;
        } else {
            return (
                <React.Fragment>
                    <span className="display-name">{`${key}: `}</span>
                    {value}
                </React.Fragment>
            );
        }
    };

    public getRenderer = (): JSX.Element => {
        return <div className="property-bag-container">{this.configs.map(this.mapper)}</div>;
    };
}
