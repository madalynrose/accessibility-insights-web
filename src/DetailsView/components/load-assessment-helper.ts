// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDataParser } from 'common/assessment-data-parser';
import { Tab } from 'common/itab';
import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';

export class LoadAssessmentHelper {
    constructor(
        private readonly assessmentDataParser: AssessmentDataParser,
        private readonly detailsViewActionMessageCreator: DetailsViewActionMessageCreator,
        private readonly fileReader: FileReader,
        private readonly document: Document,
    ) {}

    public getAssessmentForLoad(
        setAssessmentState: (versionedAssessmentData: VersionedAssessmentData) => void,
        toggleLoadAssessmentDialog: () => void,
        prevTargetPageData: Tab,
        newTargetPageId: number,
    ): void {
        const input = this.document.createElement('input');
        input.type = 'file';
        input.accept = '.a11ywebassessment';

        const onReaderLoad = (readerEvent: ProgressEvent<FileReader>) => {
            const content = readerEvent.target.result as string;
            const parsedAssessmentData = this.assessmentDataParser.parseAssessmentData(content);
            setAssessmentState(parsedAssessmentData);

            if (prevTargetPageData != null) {
                toggleLoadAssessmentDialog();
            } else {
                this.detailsViewActionMessageCreator.loadAssessment(
                    parsedAssessmentData,
                    newTargetPageId,
                );
            }
        };
        const onInputChange = (e: Event) => {
            const file = (e.target as HTMLInputElement).files[0];
            this.fileReader.onload = onReaderLoad;
            this.fileReader.readAsText(file, 'UTF-8');
        };

        input.onchange = onInputChange;
        input.click();
    }
}
