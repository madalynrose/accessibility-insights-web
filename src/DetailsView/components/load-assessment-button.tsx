// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { AssessmentDataParser } from 'common/assessment-data-parser';
import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { TabStoreData } from 'common/types/store-data/tab-store-data';
import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
import { UrlParser } from 'common/url-parser';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import { LoadAssessmentDialog } from 'DetailsView/components/load-assessment-dialog';
import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
import * as React from 'react';

export type LoadAssessmentButtonDeps = {
    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
    assessmentDataParser: AssessmentDataParser;
    urlParser: UrlParser;
    loadAssessmentHelper: LoadAssessmentHelper;
};
export interface LoadAssessmentButtonProps {
    deps: LoadAssessmentButtonDeps;
    tabStoreData: TabStoreData;
    assessmentStoreData: AssessmentStoreData;
}
export interface LoadAssessmentButtonState {
    loadedAssessmentData: VersionedAssessmentData;
    show: boolean;
}

export const loadAssessmentButtonAutomationId = 'load-assessment-button';

export class LoadAssessmentButton extends React.Component<
    LoadAssessmentButtonProps,
    LoadAssessmentButtonState
> {
    public constructor(props) {
        super(props);
        this.state = {
            loadedAssessmentData: null,
            show: false,
        };
    }

    private toggleLoadDialog = () => {
        this.setState(prevState => ({ show: !prevState.show }));
    };

    private setAssessmentState = (parsedAssessmentData: VersionedAssessmentData) => {
        this.setState(_ => ({
            loadedAssessmentData: parsedAssessmentData,
        }));
    };

    private handleLoadButtonClick = () => {
        this.props.deps.loadAssessmentHelper.getAssessmentForLoad(
            this.setAssessmentState,
            this.toggleLoadDialog,
            this.props.assessmentStoreData.persistedTabInfo,
            this.props.tabStoreData.id,
        );
    };

    public render(): JSX.Element {
        return (
            <>
                <InsightsCommandButton
                    data-automation-id={loadAssessmentButtonAutomationId}
                    iconProps={{ iconName: 'FabricOpenFolderHorizontal' }}
                    onClick={this.handleLoadButtonClick}
                >
                    Load assessment
                </InsightsCommandButton>

                <LoadAssessmentDialog
                    {...this.props}
                    tabId={this.props.tabStoreData.id}
                    loadedAssessmentData={this.state.loadedAssessmentData}
                    prevTab={this.props.assessmentStoreData.persistedTabInfo}
                    show={this.state.show}
                    onClose={this.toggleLoadDialog}
                ></LoadAssessmentDialog>
            </>
        );
    }
}
