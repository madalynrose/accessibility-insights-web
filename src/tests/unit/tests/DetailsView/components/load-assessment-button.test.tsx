// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDataParser } from 'common/assessment-data-parser';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { TabStoreData } from 'common/types/store-data/tab-store-data';
import { UrlParser } from 'common/url-parser';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import {
    LoadAssessmentButton,
    LoadAssessmentButtonProps,
    LoadAssessmentButtonDeps,
} from 'DetailsView/components/load-assessment-button';
import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
import { shallow } from 'enzyme';
import * as React from 'react';

import { Mock } from 'typemoq';

describe('LoadAssessmentButton', () => {
    const detailsViewActionMessageCreatorMock = Mock.ofType(DetailsViewActionMessageCreator);
    const assessmentDataParserMock = Mock.ofType(AssessmentDataParser);
    const urlParserMock = Mock.ofType(UrlParser);
    const loadAssessmentHelperMock = Mock.ofType(LoadAssessmentHelper);

    const tabStoreData = {
        id: 5,
    } as TabStoreData;

    const assessmentStoreData = {} as AssessmentStoreData;

    const deps = {
        detailsViewActionMessageCreator: detailsViewActionMessageCreatorMock.object,
        assessmentDataParser: assessmentDataParserMock.object,
        urlParser: urlParserMock.object,
        loadAssessmentHelper: loadAssessmentHelperMock.object,
    } as LoadAssessmentButtonDeps;
    const props = { deps, tabStoreData, assessmentStoreData } as LoadAssessmentButtonProps;

    it('should render per the snapshot', () => {
        const rendered = shallow(<LoadAssessmentButton {...props} />);
        expect(rendered.getElement()).toMatchSnapshot();
    });
});
