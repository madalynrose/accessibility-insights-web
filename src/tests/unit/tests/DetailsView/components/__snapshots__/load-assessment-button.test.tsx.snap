// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadAssessmentButton should render per the snapshot 1`] = `
<React.Fragment>
  <InsightsCommandButton
    data-automation-id="load-assessment-button"
    iconProps={
      Object {
        "iconName": "FabricOpenFolderHorizontal",
      }
    }
    onClick={[Function]}
  >
    Load assessment
  </InsightsCommandButton>
  <LoadAssessment
    assessmentStoreData={Object {}}
    deps={
      Object {
        "assessmentDataParser": proxy {
          "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
          "parseAssessmentData": [Function],
        },
        "detailsViewActionMessageCreator": proxy {
          "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
          "addPathForValidation": [Function],
          "cancelStartOver": [Function],
          "cancelStartOverAllAssessments": [Function],
          "changeAssessmentVisualizationState": [Function],
          "changeManualRequirementStatus": [Function],
          "changeManualTestStatus": [Function],
          "clearPathSnippetData": [Function],
          "closePreviewFeaturesPanel": [Function],
          "closeScopingPanel": [Function],
          "closeSettingsPanel": [Function],
          "continuePreviousAssessment": [Function],
          "copyIssueDetailsClicked": [Function],
          "dispatcher": undefined,
          "editFailureInstance": [Function],
          "leftNavPanelExpanded": [Function],
          "loadAssessment": [Function],
          "removeFailureInstance": [Function],
          "rescanVisualization": [Function],
          "saveAssessment": [Function],
          "setAllUrlsPermissionState": [Function],
          "setFeatureFlag": [Function],
          "startOverAllAssessments": [Function],
          "switchToTargetTab": [Function],
          "telemetryFactory": undefined,
          "undoManualRequirementStatusChange": [Function],
          "undoManualTestStatusChange": [Function],
        },
        "loadAssessmentHelper": proxy {
          "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
          "assessmentDataParser": undefined,
          "detailsViewActionMessageCreator": undefined,
          "document": undefined,
          "fileReader": undefined,
        },
        "urlParser": proxy {
          "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
        },
      }
    }
    loadedAssessmentData={null}
    onClose={[Function]}
    show={false}
    tabId={5}
    tabStoreData={
      Object {
        "id": 5,
      }
    }
  />
</React.Fragment>
`;
